### PR TITLE
resolve clarify borg check --verify-data (#5808)

### LINKED ISSUE #5808: docs: clarify borg check --verify-data
@andrewddavis in https://github.com/borgbackup/borg/discussions/5782#discussioncomment-734601 :
```
IMO, the man page for borg-check makes it clear that --verify-data applies
only to the archives part under the option's description:

  --verify-data: perform cryptographic archive data integrity verification (conflicts with --repository-only)

However, this could be made more clear in the main "description" section of
the man page, and the online documentation for borg, by making the paragraph
about --verify-data a bullet point of the archives check section, and including the
part about "conflicts with --repository-only".
```

### PR DESCRIPTION
**Pull request by braewoods**
This should close #5808 and count towards completion of #5129.

### CODE REVIEW

File: docs/usage/check.rst.inc [hunk 1]
```diff
@@ -107,7 +107,8 @@ First, the underlying repository data files are checked:
   the segments.
 - If checking a remote repo via ``ssh:``, the repo check is executed on the server
   without causing significant network traffic.
-- The repository check can be skipped using the ``--archives-only`` option.
+- The repository check can be skipped using the ``--archives-only`` option or the
+  ``--verify-data`` option.
```
Comment (ThomasWaldmann) [inline docs/usage/check.rst.inc:111]: not correct.

File: docs/usage/check.rst.inc [hunk 2]
```diff
@@ -149,4 +150,4 @@ checking the CRC32 of the segment) of data, which means reading the data from th
 repository, decrypting and decompressing it. This is a cryptographic verification,
 which will detect (accidental) corruption. For encrypted repositories it is
 tamper-resistant as well, unless the attacker has access to the keys. It is also very
-slow.
\ No newline at end of file
+slow. This option also implies ``--archives-only``.
```
Comment (ThomasWaldmann) [inline docs/usage/check.rst.inc:153]: not correct.

File: src/borg/archiver.py [hunk 3]
```diff
@@ -3054,7 +3054,8 @@ def define_borg_mount(parser):
           the segments.
         - If checking a remote repo via ``ssh:``, the repo check is executed on the server
           without causing significant network traffic.
-        - The repository check can be skipped using the ``--archives-only`` option.
+        - The repository check can be skipped using the ``--archives-only`` option or the
+          ``--verify-data`` option.
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3058]: not correct.

File: src/borg/archiver.py [hunk 4]
```diff
@@ -3096,7 +3097,7 @@ def define_borg_mount(parser):
         repository, decrypting and decompressing it. This is a cryptographic verification,
         which will detect (accidental) corruption. For encrypted repositories it is
         tamper-resistant as well, unless the attacker has access to the keys. It is also very
-        slow.
+        slow. This option also implies ``--archives-only``.
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3100]: not correct.

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
maybe re-read the ticket.

the repository part of the check is a very simple and low-level check. no crypto there, just some crc32 checking etc.

the crypto stuff that also needs the key is in the archives part of the check. also --verify-data happens in the archives part of the check.

if none of the --repository-only or --archives-only options are given. borg does both checks, first the lowlevel repo check, then the higher level archives check.

**Comment by braewoods** [comment]
Is this any better?

File: docs/usage/check.rst.inc [hunk 5]
```diff
@@ -125,6 +125,10 @@ Second, the consistency and correctness of the archive metadata is verified:
 - In repair mode, when all the archives were checked, orphaned chunks are deleted
   from the repo. One cause of orphaned chunks are input file related errors (like
   read errors) in the archive creation process.
+- In verify-data mode, a complete cryptographic verification of the archive data
+  integrity is performed. This conflicts with ``--repository-only`` as this mode
+  only makes sense if the archive checks are enabled. The full details of this mode
+  are documented below.
```
Comment (ThomasWaldmann) [inline docs/usage/check.rst.inc:131]: btw you do not need to update these files, that is done automatically at release time (auto-generated from archiver.py).

File: src/borg/archiver.py [hunk 6]
```diff
@@ -3072,6 +3072,10 @@ def define_borg_mount(parser):
         - In repair mode, when all the archives were checked, orphaned chunks are deleted
           from the repo. One cause of orphaned chunks are input file related errors (like
           read errors) in the archive creation process.
+        - In verify-data mode, a complete cryptographic verification of the archive data
+          integrity is performed. This conflicts with ``--repository-only`` as this mode
+          only makes sense if the archive checks are enabled. The full details of this mode
+          are documented below.
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3078]: yup, much better!

**Comment by ThomasWaldmann** [comment]
thanks!

**Pull request merged**